### PR TITLE
feat(ereal): standardize math/functions fractional/error_gamma + smoke (#950 cat 5, final)

### LINKED ISSUE #950: Refactor elastic/ereal/math/functions/ (16 files) to standard test pattern
**Parent epic: #944**

## Goal

Refactor the 16 files under `elastic/ereal/math/functions/` to the standard regression-test structure. One function (or tightly related family) per file.

## Files

- classify.cpp -- isnan, isinf, isnormal, fpclassify
- ereal_math_stub_test.cpp -- decide if obsolete; possibly remove
- error_and_gamma.cpp -- erf, erfc, lgamma, tgamma
- exponent.cpp -- exp, exp2, expm1, exp10
- fractional.cpp -- modf, fmod, remainder
- hyperbolic.cpp -- sinh, cosh, tanh, asinh, acosh, atanh
- hypot.cpp -- hypot
- logarithm.cpp -- log, log2, log10, log1p
- minmax.cpp -- min, max
- next.cpp -- nextafter, nexttoward
- numerics.cpp -- abs, copysign, signbit
- pow.cpp -- pow, sqrt-by-pow special cases
- sqrt.cpp -- sqrt, cbrt
- trigonometry.cpp -- sin, cos, tan, asin, acos, atan, atan2
- truncate.cpp -- ceil, floor, trunc, round, rint

## Pattern

See `arithmetic/addition.cpp` post-PR #943. Each file gets the `MANUAL_TESTING` / `REGRESSION_LEVEL_OVERRIDE` scaffold and uses `ReportTestSuiteHeader / ReportTestResult / ReportTestSuiteResults`.

## Coverage per function

Verify against high-precision reference values (MPFR oracle when #944's Phase E lands) for representative inputs. Pre-fuzzer, hand-curated test cases:
- Identity / fixed points (e.g. `exp(0) == 1`, `log(1) == 0`)
- Inverse relations (e.g. `exp(log(x)) == x`)
- Sign symmetries (e.g. `sin(-x) == -sin(x)`)
- Special values (`+0`, `-0`, `inf`, `nan` where applicable)
- Subnormal-input edge cases

## Acceptance

- All 16 files use the standard framework.
- Each test is single-function (function under test only; setup may use constructors / unrelated math).
- Build clean gcc + clang, debug + release.
- `ereal_math_stub_test.cpp` decision documented (keep, rename, or remove).

## Suggested decomposition

This issue is large -- consider splitting into one PR per file or one PR per category (trig, exp/log, hyperbolic, etc.). Reviewer load is the main constraint.

### PR DESCRIPTION
**Pull request by Ravenwater**
## Summary
Final category (5 of 5) for the `elastic/ereal/math/functions/` refactor: `fractional.cpp`, `error_and_gamma.cpp`, `ereal_math_stub_test.cpp`. **Completes the 15-file standardization.**

Relates to #944 (parent epic, Phase B). **Resolves #950.**

## Changes
- **`fractional.cpp`** -- standardized; kept the fmod/remainder/div-by-zero/fmod-vs-remainder tests; added `VerifyFractionalFuzz` (reduction identity `x==trunc(x/y)*y+fmod(x,y)`, `|fmod|<|y|`, fmod sign, `|remainder|<=|y|/2`). `modf` is **not** covered -- it is not implemented for `ereal` (only `std::modf` is used internally).
- **`error_and_gamma.cpp`** -- was a Phase-0 `MANUAL_TESTING` smoke stub; erf/erfc/tgamma/lgamma turn out to be implemented and accurate, so converted to real tests: `VerifyErrorFunction` (erf(0)==0, vs std, parity, `erf+erfc==1`), `VerifyTgamma` (`tgamma(n)==(n-1)!`, `tgamma(1/2)^2==pi`, recurrence `tgamma(x+1)==x*tgamma(x)`), `VerifyLgamma` (fixed points, `lgamma==log(tgamma)`, vs std), and a property fuzzer.
- **`ereal_math_stub_test.cpp`** -- refactored the print-only demo into `VerifyMathSmoke`, a breadth-first regression that calls every mathlib function once on an in-domain input and asserts a finite result (kept, not removed, per maintainer preference; complements the per-function files).

## Test Results
| Target | gcc build | gcc test | clang build | clang test |
|--------|-----------|----------|-------------|------------|
| er_math_fractional | OK | PASS | OK | PASS |
| er_math_error_and_gamma | OK | PASS | OK | PASS |
| er_math_ereal_math_stub_test | OK | PASS | OK | PASS |

## #950 complete
cat1 (#974) · cat2 (#975) · cat3 (#976) · cat4 (#977) · cat5 (this) -- all 15 files standardized to the addition.cpp framework with oracle-free property/identity coverage. The precision-lifting self-oracle (and the `log1p` precision gap surfaced in cat2) remain scoped to #954.

## Test plan
- [ ] Fast CI passes (gcc + clang CI_LITE)
- [ ] Promote to ready when satisfied: `gh pr ready <NNN>`

Resolves #950

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Converted manual smoke tests into automated regression tests integrated with the project test runner.
  * Added comprehensive verifiers for math families: classification, arithmetic, truncation, roots, exponentials/logs, trig, error/gamma, and nextafter.
  * Introduced deterministic and randomized property-based fuzzers to broaden coverage.
  * Standardized reporting and simplified exception handling; adjusted manual-testing defaults to enable automated runs.

[Review Change Stack](https://app.coderabbit.ai/change-stack/stillwater-sc/universal/pull/978?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: elastic/ereal/math/functions/error_and_gamma.cpp [hunk 1]
```diff
@@ -1,27 +1,160 @@
...
+				if (reportTestCases) std::cout << "    FAIL erf monotonic at " << e << "\n"; ++nrOfFailedTestCases;
+			}
+			double g = gdist(rng);
+			Real y(g);
+			if (!close_rel(tgamma(y + Real(1.0)), y * tgamma(y), 1.0e-12)) {
+				if (reportTestCases) std::cout << "    FAIL gamma recurrence at " << g << "\n"; ++nrOfFailedTestCases;
+			}
+		}
+		return nrOfFailedTestCases;
+	}
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply elastic/ereal/math/functions/error_and_gamma.cpp:172]: Addressed in the latest commit (verified gcc-13 + clang-18).

File: elastic/ereal/math/functions/fractional.cpp [hunk 2]
```diff
@@ -223,15 +226,54 @@ namespace sw {
...
+				double recon = double(n * y + r);
+				if (std::abs(recon - dx) > 1.0e-10 * (1.0 + std::abs(dx))) {
+					if (reportTestCases) std::cout << "    FAIL fmod reduction at x=" << dx << " y=" << dy << '\n';
+					++nrOfFailedTestCases;
+				}
+				// |fmod(x,y)| < |y| and same sign as x (when nonzero)
+				if (std::abs(double(r)) > std::abs(dy)) {
+					if (reportTestCases) std::cout << "    FAIL |fmod| >= |y| at x=" << dx << '\n';
+					++nrOfFailedTestCases;
+				}
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply elastic/ereal/math/functions/fractional.cpp]: Addressed in the latest commit (verified gcc-13 + clang-18).

File: elastic/ereal/math/functions/error_and_gamma.cpp [hunk 3]
```diff
@@ -1,27 +1,188 @@
...
+			if (!close_rel(erf(-x), -erf(x), 1.0e-14)) {
+				if (reportTestCases) std::cout << "    FAIL erf parity at " << v << "\n"; ++nrOfFailedTestCases;
+			}
+			if (!close_rel(erf(x) + erfc(x), Real(1.0), 1.0e-14)) {
+				if (reportTestCases) std::cout << "    FAIL erf+erfc != 1 at " << v << "\n"; ++nrOfFailedTestCases;
+			}
+		}
+		// special values: erf(+/-Inf) = +/-1, erfc(+/-Inf) = 0/2, NaN propagates
+		double pinf = std::numeric_limits<double>::infinity();
+		double qnan = std::numeric_limits<double>::quiet_NaN();
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply elastic/ereal/math/functions/error_and_gamma.cpp:64]: Fixed -- added explicit `#include <limits>`. Verified gcc-13 + clang-18.

**Comment by Ravenwater** [comment]
@coderabbitai help

**Comment by Ravenwater** [comment]
@coderabbitai rate limit

**Comment by Ravenwater** [comment]
@coderabbitai configuration
